docs/source/extension/extension_migration.rst

```diff
@@ -189,6 +190,7 @@ bumped their major version (following semver convention). We want to point out p
 - ``@jupyterlab/notebook`` from 3.x to 4.x
    * The ``NotebookPanel._onSave`` method is now ``private``.
    * ``NotebookActions.collapseAll`` method renamed to ``NotebookActions.collapseAllHeadings``.
+   * ``NotebookActions.run`` and other run commands now require ``ISessionContext``.
```

Comment:
This does seem to be accurate anymore according to the latest state of the diff?

```suggestion
```

Reply:
It is accurate - only the UI related elements (dialogs and translator) are optional:

![image](https://user-images.githubusercontent.com/8435071/224049621-ed8f717a-2cbc-425e-8624-2dc5cd6b8cd3.png)

Reply:
ah I thought we could get around without requiring a `ISessionContext` as this level as well.

Reply:
Why is it required in `NotebookActions.run` here:

https://github.com/jupyterlab/jupyterlab/blob/d3ae4678aca111b209e392ed0914cb565b7c4ee3/packages/notebook/src/actions.tsx#L502-L507

If it is not in the underlying `Private.runSelected` here:

https://github.com/jupyterlab/jupyterlab/blob/d3ae4678aca111b209e392ed0914cb565b7c4ee3/packages/notebook/src/actions.tsx#L2149-L2154

(links taken from the latest state of this PR)